linodecli/plugins/k8s-alpha.py

```diff
@@ -0,0 +1,286 @@
+                             '(default 3)')
+    parser.add_argument('--master-type', metavar="TYPE", type=str, required=False,
+                        default=tf_var_map['master_type']['default'],
+                        help='Linode Type ID for cluster Master Nodes as retrieved with '
+                             '`linode-cli linodes types`. (default "g6-standard-2")')
+    parser.add_argument('--region', metavar="REGION", type=str, required=False,
+                        default=tf_var_map['region']['default'],
+                        help='The Linode Region ID in which to deploy the cluster as retrieved with '
+                             '`linode-cli regions list`. (default "us-west")')
+    parser.add_argument('--ssh-private-key', metavar="KEYPATH", type=str, required=False,
+                        default=tf_var_map['ssh_private_key']['default'],
+                        help='The path to your private key file which will be used to access Nodes')
+    parser.add_argument('--ssh-public-key', metavar="KEYPATH", type=str, required=False,
+                        default=tf_var_map['ssh_public_key']['default'],
+                        help='The path to your public key file which will be used to access Nodes')
+    parsed, _ = parser.parse_known_args(args)
+
+    # Check if Terraform is installed
+    try:
+        nullf = open(os.devnull, 'w')
+        spcall(['terraform'], stdout=nullf)
+    except:
+        print('To create a cluster you must first install Terraform\n'
+              'This command will automatically download and install the Linode provider '
+              'for Terraform.\n'
+              '\nOn macOS with Homebrew: brew install terraform\n'
+              'For other platforms, use your package manager and/or refer to this documentation\n'
+              'https://learn.hashicorp.com/terraform/getting-started/install.html')
+        sys.exit(1)
+
+    hashname = get_hashname(parsed.name)
+#   MAJOR @TODO: check here if this hashname already appears as a prefix on any
+#   Volumes, Linodes, or NodeBalancers. If it does, bail with an error message,
+#   because we don't want to later delete resources from multiple clusters!
+
+    # Make application directory if it doesn't exist
+    appdir = safe_make_appdir("k8s-alpha")
+    # Make the terraform project directory if it doesn't exist
+    terrapath = os.path.join(appdir, parsed.name)
+    safe_mkdir(terrapath)
+
+    # Move to the directory
+    os.chdir(terrapath)
+
+    # Generate the terraform file
+    terrafile = open('cluster.tf', 'w')
+    terrafile.write(gen_terraform_file(context, parsed.name, hashname))
+    terrafile.close()
+
+    # Generate terraform args
+    terraform_args = gen_terraform_args(parsed)
+
+    # Run the Terraform commands
+    spcall(['terraform', 'workspace', 'new', parsed.name])
+    call_or_exit(['terraform', 'init'])
+    # TODO: Before running the apply delete any existing Linodes that would
+    # cause the apply to fail.
```

Comment:
I'm not familiar with Terraform - in what case would existing Linodes cause this apply to fail?

Reply:
In some cases I've seen Terraform fail to provision a Linode and not have it stored in state (to be re-provisioned on a subsequent run). In this case the user is stuck until the manually delete the Linode. It's rare.